src/Lecture2.hs

```diff
@@ -101,7 +112,20 @@ spaces.
 
 🕯 HINT: look into Data.Char and Prelude modules for functions you may use.
 -}
-dropSpaces = error "TODO"
+dropSpaces :: [Char] -> [Char]
+-- dropSpaces = filter (not . isSpace)  
+-- the function above works, but not on infinite trailing spaces
+
+-- following solves this, but no eta-reduction or function composition used
+--   so there are other, probably shorter/better ways ...
+--
+-- dropSpaces list = headSave (filter (\(x:_) -> not (isSpace x)) (words list))
```

Comment:
The `\(x:_)` is a partial pattern here. Generally, it's better to avoid such patterns. It's hard to see from the implementation if the list is always non-empty in the input 🤔